cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlServletApp.java

```diff
@@ -120,12 +122,14 @@ private void maybeConfigureTlsProtocolsAndCiphers(SslContextFactory sslContextFa
     protected void setupWebUi(ServletContextHandler contextHandler) {
         // Placeholder for any static content
         String webuiDir = _config.getString(WebServerConfig.WEBSERVER_UI_DISKPATH_CONFIG);
-        String webuiPathPrefix = _config.getString(WebServerConfig.WEBSERVER_UI_URLPREFIX_CONFIG);
-        DefaultServlet defaultServlet = new DefaultServlet();
-        ServletHolder holderWebapp = new ServletHolder("default", defaultServlet);
-        // holderWebapp.setInitParameter("org.eclipse.jetty.servlet.Default.dirAllowed", "false");
-        holderWebapp.setInitParameter("resourceBase", webuiDir);
-        contextHandler.addServlet(holderWebapp, webuiPathPrefix);
+        Path path = Path.of(webuiDir);
+        if (Files.isDirectory(path) && Files.isReadable(path)) {
+            String webuiPathPrefix = _config.getString(WebServerConfig.WEBSERVER_UI_URLPREFIX_CONFIG);
+            DefaultServlet defaultServlet = new DefaultServlet();
+            ServletHolder holderWebapp = new ServletHolder("default", defaultServlet);
+            contextHandler.setBaseResourceAsString(webuiDir);
+            contextHandler.addServlet(holderWebapp, webuiPathPrefix);
+        }
```

Comment:
> Need to add an if statement here. When webuiDir is null an IAE is thrown and causes tests to fail

Nice catch, it might also be worth logging a warning when the webuiDir is not found, something like this:
```
else {
    LOG.warn("WebUI directory not found or unreadable: {} UI disabled", webuiDir);
}
```
This way, the UI won't fail silently.